eva-accession-core/src/test/java/uk/ac/ebi/eva/accession/core/SubmittedVariantAccessioningServiceTest.java

```diff
@@ -104,7 +107,71 @@ public void sameAccessionsAreReturnedForEquivalentVariants() throws AccessionCou
                 requestedVariants);
 
-        assertEquals(new HashSet<>(generatedAccessions),
-                     new HashSet<>(retrievedAccessions));
+        assertEquals(new HashSet<>(generatedAccessions), new HashSet<>(retrievedAccessions));
+    }
+
+    @Test
+    public void getOrCreateAccessions() throws AccessionCouldNotBeGeneratedException {
+        List<SubmittedVariant> variants = Arrays.asList(
+                new SubmittedVariant("GCA_000003055.3", 9913, "PRJEB21794", "21", 20800319, "C", "T", CLUSTERED_VARIANT,
+                                     SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH, VALIDATED),
+                new SubmittedVariant("assembly", 1111, "project", "contig_2", 100, "ref", "alt", CLUSTERED_VARIANT,
+                                     SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH, VALIDATED),
+                new SubmittedVariant("GCA_000009999.3", 9999, "PRJEB21999", "21", 20849999, "", "GG", CLUSTERED_VARIANT,
+                                     SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, true, VALIDATED));
+
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = service.getOrCreate(variants);
+
+        long accession = submittedVariants.stream().filter(x -> x.getData().getProjectAccession().equals("PRJEB21794"))
+                                          .collect(Collectors.toList())
+                                          .get(0).getAccession();
+        assertEquals(5000000000L, accession);
+
+        long accessionDbsnp = submittedVariants.stream()
+                                               .filter(x -> x.getData().getProjectAccession().equals("PRJEB21999"))
+                                               .collect(Collectors.toList())
+                                               .get(0).getAccession();
+        assertEquals(1000000001L, accessionDbsnp);
+
+        assertEquals(3, submittedVariants.size());
+    }
+
+    @Test
+    public void get() {
+        List<SubmittedVariant> variants = Arrays.asList(
+                new SubmittedVariant("GCA_000003055.3", 9913, "PRJEB21794", "21", 20800319, "C", "T", CLUSTERED_VARIANT,
+                                     SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH, VALIDATED),
+                new SubmittedVariant("assembly", 1111, "project", "contig_2", 100, "ref", "alt", CLUSTERED_VARIANT,
+                                     SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH, VALIDATED),
+                new SubmittedVariant("GCA_000009999.3", 9999, "PRJEB21999", "21", 20849999, "", "GG", CLUSTERED_VARIANT,
+                                     SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, true, VALIDATED));
+
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> accessions = service.get(variants);
+        assertEquals(2, accessions.size());
+    }
+
+    @Test
+    public void getByAccessions() {
```

Comment:
If you don't annotate test methods as
```java
@UsingDataSet(loadStrategy = LoadStrategyEnum.DELETE_ALL)
```
the same dataset will be used across multiple tests, creating dependencies and potentially causing bugs. See how the first methods in the class have said annotation.